File: JAVA/Extract the Number from the String.java
class Solution {
    long ExtractNumber(String sentence) {
        Pattern pattern = Pattern.compile("\\b\\d+\\b");
        Matcher matcher = pattern.matcher(sentence);
        long maxNum = -1;
        while (matcher.find()) {
            long num = Long.parseLong(matcher.group());
            if (!String.valueOf(num).contains("9") && num > maxNum) {
                maxNum = num;
            }
        }
        return maxNum;
    }
}

File: C++/Extract the Number from the String.cpp
class Solution {
  public:
    long long ExtractNumber(string sentence) {
        long long int result = 0;
        bool flag = 0;
        bool has9 = 0;
        long long int temp = 0;
        for(char i:sentence){
            if (i>=48 && i<=57 && !flag)
            {
                flag = 1;
                has9 = 0;
                temp = 0;
            }
            else if(i<48 || i>57 && flag){
                flag = 0;
                if(!has9 && temp > result) result = temp;
            }
            if(flag) {
                if (i == 57) has9 = 1;
                temp = temp*10 + i%48;
            }
        }
        if (!has9 && temp > result) result = temp;
        return (result == 0)? -1 : result;
    }
};
